int64validator/at_least_sum_of.go

```diff
@@ -0,0 +1,106 @@
+package int64validator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+)
+
+var _ tfsdk.AttributeValidator = atLeastSumOfValidator{}
+
+// atLeastSumOfValidator validates that an integer Attribute's value is at least the sum of one
+// or more integer Attributes retrieved via the given path expressions.
+type atLeastSumOfValidator struct {
+	attributesToSumPathExpressions path.Expressions
+}
+
+// Description describes the validation in plain text formatting.
+func (validator atLeastSumOfValidator) Description(_ context.Context) string {
+	var attributePaths []string
+	for _, p := range validator.attributesToSumPathExpressions {
+		attributePaths = append(attributePaths, p.String())
+	}
+
+	return fmt.Sprintf("value must be at least sum of %s", strings.Join(attributePaths, " + "))
+}
+
+// MarkdownDescription describes the validation in Markdown formatting.
+func (validator atLeastSumOfValidator) MarkdownDescription(ctx context.Context) string {
+	return validator.Description(ctx)
+}
+
+// Validate performs the validation.
+func (validator atLeastSumOfValidator) Validate(ctx context.Context, request tfsdk.ValidateAttributeRequest, response *tfsdk.ValidateAttributeResponse) {
+	i, ok := validateInt(ctx, request, response)
+
+	if !ok {
+		return
+	}
+
+	var sumOfAttribs int64
+	var numUnknownAttribsToSum int
+
+	for _, expression := range validator.attributesToSumPathExpressions {
+		matchedPaths, diags := request.Config.PathMatches(ctx, expression)
+		response.Diagnostics.Append(diags...)
+
+		// Collect all errors
+		if diags.HasError() {
+			continue
+		}
+
+		for _, mp := range matchedPaths {
+			var attribToSum types.Int64
+
+			diags := request.Config.GetAttribute(ctx, mp, &attribToSum)
+			response.Diagnostics.Append(diags...)
+
+			// Collect all errors
+			if diags.HasError() {
+				continue
+			}
+
+			if attribToSum.IsNull() {
+				continue
+			}
+
+			if attribToSum.IsUnknown() {
+				numUnknownAttribsToSum++
+				continue
+			}
+
+			sumOfAttribs += attribToSum.Value
+		}
+	}
+
+	if numUnknownAttribsToSum == len(validator.attributesToSumPathExpressions) {
+		return
+	}
```

Comment:
I see what you are doing here: fi they are all unknown, skip validation and don't error.

The issue I see here is that in case _some_ are known, but not _all_, you will then reach the `i < sumOfAttribs`, and that can fail and return an error. But if when they all turn _known_, the sum matches the validation expectation, then it would pass.

Isn't it better to just skip once you hit an unknown? It might be _blunt_, but it's less confusing to practitioners: this is how @bflad I think made me approach the `schemavalidator`s.

Reply:
Good point.
Have amended to not validate if any attributes are unknown.